### PR TITLE
Add option to allow package upgrades.

### PR DESCRIPTION
**Pull request by fjgarlin**
Screenshot 2022-06-08 at 17 36 02

Running the composer commands on the lastest Drupal version fails for `drupal/core-dev`. Adding the `-W` fixes the issue.

### CODE REVIEW

**Comment by fjgarlin** [comment]
`-W` requires composer 2. Checking options.

**Comment by fjgarlin** [comment]
@juampynr - as seen on the screenshot, it was failing for the latest D9 version. 

I added the `-W` option to the composer command, which also needs composer 2 (the image seems to be using composer 1).

File: setup-bitbucket.sh [hunk 1]
```diff
@@ -25,9 +25,11 @@ drupal9ci_install() {
 	rsync -va --ignore-existing "$tmpdir/drupal9ci-master/dist/common/" .
 
 	# Add development dependencies to run the jobs.
+	composer self-update --2
 	COMPOSER_MEMORY_LIMIT=-1 composer require --dev	\
 		drupal/core-dev \
-		drupal/drupal-extension
+		phpspec/prophecy-phpunit:^2 \
+		drupal/drupal-extension -W
```
Comment (juampynr) [inline setup-bitbucket.sh:32]: LOL, I had no idea what -W was. I think that I used it before but it was the long option name.

**Review by juampynr — APPROVED** [review]
Approving since this looks good so far.

File: dist/bitbucket/RoboFile.php [hunk 2]
```diff
@@ -164,8 +164,6 @@ protected function runCoverageReport() {
    */
   protected function runCodeSniffer() {
     $tasks = [];
-    $tasks[] = $this->taskExecStack()
```
Comment (fjgarlin) [inline dist/bitbucket/RoboFile.php:167]: `coder` > 8.1.3 requires NOT to set this up.

File: dist/bitbucket/RoboFile.php [hunk 3]
```diff
@@ -185,6 +183,7 @@ function runServeDrupal()
     $tasks = [];
     $tasks[] = $this->taskExec('chown -R www-data:www-data ' . getenv('BITBUCKET_CLONE_DIR'));
     $tasks[] = $this->taskExec('ln -sf ' . getenv('BITBUCKET_CLONE_DIR') . '/web /var/www/html');
+    $tasks[] = $this->taskExec('echo "\nServerName localhost" >> /etc/apache2/apache2.conf');
```
Comment (fjgarlin) [inline dist/bitbucket/RoboFile.php:186]: Not sure why but localhost wasn't the default, so forcing it.

File: dist/bitbucket/RoboFile.php [hunk 4]
```diff
@@ -220,7 +219,7 @@ protected function runCypressTests()
       ->copy('.cypress/cypress.json', 'cypress.json', $force)
       ->copy('.cypress/package.json', 'package.json', $force);
     $tasks[] = $this->taskExec('sleep 30s');
-    $tasks[] = $this->taskExec('npm install cypress --save-dev');
+    $tasks[] = $this->taskExec('npm install cypress@9 --save-dev');
```
Comment (fjgarlin) [inline dist/bitbucket/RoboFile.php:222]: Cypress released a major new version in May 2022 which breaks compatibility, so fixing version.

File: dist/circleci/.circleci/RoboFile.php [hunk 5]
```diff
@@ -148,6 +148,9 @@ protected function runBehatTests()
     {
         $force = TRUE;
         $tasks = [];
+        $tasks[] = $this->taskExec('chown -R www-data:www-data /opt/drupal');
```
Comment (fjgarlin) [inline dist/circleci/.circleci/RoboFile.php:151]: Needed as otherwise we get permission denied.

File: dist/travisci/.travis.yml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 language: php
-dist: xenial
+dist: bionic
```
Comment (fjgarlin) [inline dist/travisci/.travis.yml:2]: sqlite version for php 7.4 is very old and xenial didn't play well updating it.

File: dist/travisci/.travis.yml [hunk 7]
```diff
@@ -24,8 +24,12 @@ env:
 
 before_install:
   - echo 'sendmail_path = /bin/true' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
+  - sudo echo "deb http://archive.ubuntu.com/ubuntu focal main restricted universe multiverse" >> /etc/apt/sources.list
+  - sudo apt-get update
+  - sudo apt-get -y install sqlite3/focal
```
Comment (fjgarlin) [inline dist/travisci/.travis.yml:29]: sqlite version for php 7.4 is very old and Drupal 9 requires a newer one.

File: dist/travisci/.travis/RoboFile.php [hunk 8]
```diff
@@ -235,7 +235,7 @@ protected function runUnitTests()
             ->copy('.travis/config/phpunit.xml', 'web/core/phpunit.xml', $force);
         $tasks[] = $this->taskExecStack()
             ->dir('web')
-            ->exec('../vendor/bin/phpunit -c core --debug --coverage-clover ../build/logs/clover.xml --verbose modules/custom');
```
Comment (fjgarlin) [inline dist/travisci/.travis/RoboFile.php:238]: getting segmentation fault and nesting level limits reached with xdebug. not requiring it to run the test.

File: setup-bitbucket.sh [hunk 9]
```diff
@@ -25,9 +25,12 @@ drupal9ci_install() {
 	rsync -va --ignore-existing "$tmpdir/drupal9ci-master/dist/common/" .
 
 	# Add development dependencies to run the jobs.
+	composer self-update --2
 	COMPOSER_MEMORY_LIMIT=-1 composer require --dev	\
 		drupal/core-dev \
-		drupal/drupal-extension
+		drupal/coder \
```
Comment (fjgarlin) [inline setup-bitbucket.sh:31]: Needs to be explicit to set it up properly.

**Comment by fjgarlin** [comment]
Travis and Circle down and fully tested. I might create some follow up issues, but this PR is to try to get it to work with latest versions and the images provided.

Will get to the other three platforms within the next days and it'll hopefully be quicker/easier. I'll also squash the commit to avoid a history of tests and switching on/off things.

File: dist/travisci/.travis.yml (hunk 7 above)
Comment (fjgarlin) [reply dist/travisci/.travis.yml]: Changed the approach to run things inside the docker container so no longer needed.

**Comment by fjgarlin** [comment]
@juampynr - I tested all the changes on all five platforms and it's all good. I'll wait until tonight to merge, but not many changes since you approved.

**Comment by juampynr** [comment]
> @juampynr - I tested all the changes on all five platforms and it's all good. I'll wait until tonight to merge, but not many changes since you approved.

Just saw them. Everything looks fine!

**Pull request merged**